m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -380,37 +365,34 @@ public void reportHistogramDurationSamples(
 
         String bucketIdFmt = String.format("%%0%sd", bucketIdLen);
 
-        BucketPair[] bucketPairs = BucketPairImpl.bucketPairs(buckets);
+        ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<>();
```

Comment:
mind dropping a TODO here that this can be optimized to avoid repeated builder creations in the future?

Reply:
How do you want to optimize this? This Map is not static

Reply:
talked offline: we can address if it comes up in future performance profiles